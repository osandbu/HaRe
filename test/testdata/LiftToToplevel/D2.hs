module LiftToToplevel.D2 where

{-lift 'sq' to top level. In this refactoring, 'sq' will
  be hided in the import declaraion of module 'D2' in module 'C2'.-}
 
sumSquares (x:xs) = sq x + sumSquares xs
  where 
     sq x = x ^ pow
     pow =2 
 
sumSquares [] = 0

