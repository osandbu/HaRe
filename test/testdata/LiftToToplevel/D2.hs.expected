module LiftToToplevel.D2 where

{-lift 'sq' to top level. In this refactoring, 'sq' will
  be hided in the import declaraion of module 'D2' in module 'C2'.-}

sumSquares (x:xs) = (sq pow)x + sumSquares xs
  where

     pow =2

sumSquares [] = 0

sq pow x= x ^ pow

